tensorboard/util.py

```diff
@@ -80,6 +81,18 @@ def closeable(class_):
   return class_
 
 
+def guarded_by(unused_field):
+  """Indicates method should be called from within a lock.
+
+  This decorator is purely for documentation purposes. It has the same
```

Comment:
That could work, but I'm not sure off the top of my head how I would pass the field itself to the decorator. Wouldn't it have to be a class level variable? I guess it could be a string and it uses getattr(self, "field"). If you want to do that in a follow-up, I wouldn't be opposed.